ПР7

Megrief opened the pull request:
SearchActivity:
- Поиск производится с задержкой в 2 секунды после окончания ввода текста
- Добавлен ProgressBar 
- Добавлен debouncer для блокировки множественных нажатий
AudioplayerActivity:
- Добавлена возможность воспроизведения аудио-фрагмента
- Кнопка play-pause стала функциональной и изменяется в зависимости от состояния воспроизведения
- Добавлен таймер показывающий прогресс воспроизведения

Megrief committed:
Передача Track в AudioplayerActivity:
Shared preferences replaced with Intent.

Megrief committed:
Доработана SearchActivity:
- Отложенный поиск (SearchRequestDebouncer)
- Блокировка множественных нажатий на элементы списка (ItemClickDebouncer)
- Добавлен ProgressBar

Megrief committed:
Доработана AudioplayerActivity:
- воспроизведение фрагмента аудио
- обновление времени воспроизведения

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/activities/SearchActivity.kt:
```diff
@@ -36,24 +36,56 @@ import retrofit2.converter.gson.GsonConverterFactory
+    private val onTrackClicked: (Track) -> Unit = {
+        if (itemClickDebouncer.clickDebounce()) {
+            searchHistory.addToHistory(it)
+            val jTrack = Gson().toJson(it, Track::class.java)
```
Не экономь на буквах, лучше написать jsonTrack

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/activities/SearchActivity.kt:
```diff
@@ -36,24 +36,56 @@ import retrofit2.converter.gson.GsonConverterFactory
+            val jTrack = Gson().toJson(it, Track::class.java)
+            startActivity(
+                Intent(this, AudioplayerActivity::class.java)
+                    .putExtra(K_TRACK, jTrack)
```
K_TRACK - трудно понять что тут за ключ должен храниться

AleksandrIlinskii commented on app/src/main/java/com/example/playlistmaker/utils/Player.kt:
```diff
@@ -0,0 +1,71 @@
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Player(url: String,
```
Если пишешь параметры в несколько строк, то нужно тут оставить открытую скобку, и каждый новый аргумент с новой строки, а в конце последнего снова ударить энтером

AleksandrIlinskii commented:
Расположение длительности трека не соответствует макетам
<img width="570" alt="image" src="https://github.com/Megrief/PlaylistMaker/assets/129396632/b6a8dc70-9f56-4a41-a9ef-76001556dcca">

Megrief committed:
Исправлены недочеты:
- Оформление конструкторов классов
- Имена констант и переменных
- Верстка track_card в соответствие с макетом
